app/src/org/commcare/CommCareHeartbeatManager.java

```diff
@@ -0,0 +1,309 @@
+ * with a "heartbeat" request, and then handle the response. The lifecycle of the TimerTask is
+ * tied to that of the CommCareSessionService; it should be started whenever a session service is
+ * started, and ended whenever a session service is ended for any reason.
+ *
+ * Currently, the primary content of the server's response to the heartbeat request will be
+ * information about potential binary or app updates that the app should prompt users to conduct.
+ *
+ * Created by amstone326 on 4/13/17.
+ */
+public class CommCareHeartbeatManager {
+
+    private static final long ONE_DAY_IN_MS = 24 * 60 * 60 * 1000;
+
+    private static final String TEST_RESPONSE =
+            "{\"latest_apk_version\":{\"value\":\"2.36.1\"},\"latest_ccz_version\":{\"value\":\"197\", \"force_by_date\":\"2017-04-24\"}}";
+
+    private static final String QUARANTINED_FORMS_PARAM = "num_quarantined_forms";
+    private static final String UNSENT_FORMS_PARAM = "num_unsent_forms";
+    private static final String LAST_SYNC_TIME_PARAM = "last_sync_time";
+
+    private Timer heartbeatTimer;
+
+    private final HttpResponseProcessor responseProcessor = new HttpResponseProcessor() {
+
+        @Override
+        public void processSuccess(int responseCode, InputStream responseData) {
+            try {
+                String responseAsString = StreamsUtil.inputStreamToByteArray(responseData).toString();
+                JSONObject jsonResponse = new JSONObject(responseAsString);
+                parseHeartbeatResponse(jsonResponse);
+            }
+            catch (JSONException e) {
+                Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                        "Heartbeat response was not properly-formed JSON: " + e.getMessage());
+            }
+            catch (IOException e) {
+                Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                        "IO error while processing heartbeat response: " + e.getMessage());
+            }
+        }
+
+        @Override
+        public void processRedirection(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void processClientError(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void processServerError(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void processOther(int responseCode) {
+            processErrorResponse(responseCode);
+        }
+
+        @Override
+        public void handleIOException(IOException exception) {
+            Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
+                    "Encountered IOException while getting response stream for heartbeat response: "
+                            + exception.getMessage());
+        }
+
+        private void processErrorResponse(int responseCode) {
+            Logger.log(AndroidLogger.TYPE_ERROR_SERVER_COMMS,
```

Comment:
We may end up processing a lot of these, so I think we should be cautious to avoid logging errors to our forensics lots unless we can explicitly rate limit them.